crawl-ref/source/travel.cc

```diff
@@ -2214,6 +2219,126 @@ static int _prompt_travel_branch(int prompt_flags)
     }
 }
 
+static god_type _get_god_by_initial(const char god_initial)
+{
+    switch (toupper_safe(god_initial))
+    {
+        case '1': return GOD_SHINING_ONE;
+        case 'A': return GOD_ASHENZARI;
+        case 'B': return GOD_BEOGH;
+        case 'C': return GOD_CHEIBRIADOS;
+        case 'D': return GOD_DITHMENOS;
+        case 'E': return GOD_ELYVILON;
+        case 'F': return GOD_FEDHAS;
+        case 'G': return GOD_GOZAG;
+        case 'H': return GOD_HEPLIAKLQANA;
+        case 'J': return GOD_JIYVA;
+        case 'K': return GOD_KIKUBAAQUDGHA;
+        case 'L': return GOD_LUGONU;
+        case 'M': return GOD_MAKHLEB;
+        case 'N': return GOD_NEMELEX_XOBEH;
+        case 'O': return GOD_OKAWARU;
+#if TAG_MAJOR_VERSION == 34
+        case 'P': return GOD_PAKELLAS;
+#endif
+        case 'Q': return GOD_QAZLAL;
+        case 'R': return GOD_RU;
+        case 'S': return GOD_SIF_MUNA;
+        case 'T': return GOD_TROG;
+        case 'U': return GOD_USKAYAW;
+        case 'V': return GOD_VEHUMET;
+        case 'W': return GOD_WU_JIAN;
+        case 'X': return GOD_XOM;
+        case 'Y': return GOD_YREDELEMNUL;
+        case 'Z': return GOD_ZIN;
+        default:  return GOD_NO_GOD;
+    }
+}
+
+static level_pos _prompt_travel_altar()
+{
+    extern map<level_pos, god_type> altars_present;
+
+    if (altars_present.empty())
+        return level_pos();
+
+    level_pos nearest_altars[NUM_GODS];
+    int best_altar_dist = -1;
+    const level_id curr = level_id::current();
+
+    // Populate nearest_altars[] with nearest altars
+    for (const auto &entry : altars_present)
+    {
+        if (entry.second >= NUM_GODS)
```

Comment:
Is this check necessary? Perhaps it should be an assert?

Reply:
As discussed on ##crawl-dev, this is necessary because of the faded altar (i.e., GOD_ECUMENICAL). Comment added to improve code clarity.